optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/view/IndictmentView.java

```diff
@@ -0,0 +1,87 @@
+import org.optaplanner.core.api.score.constraint.Indictment;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+public class IndictmentView {
+
+    private List<Rules> ruleMatchedList;
+    private List<HardSoftScore> scoreList;
+
+    public IndictmentView() {
+
+    }
+
+    public IndictmentView(Indictment indictment) {
+        ruleMatchedList = new ArrayList<>(indictment.getConstraintMatchCount());
+        scoreList = new ArrayList<>(indictment.getConstraintMatchCount());
+
+        for (ConstraintMatch cm : indictment.getConstraintMatchSet()) {
+            ruleMatchedList.add(Rules.forConstraintMatch(cm));
+            scoreList.add((HardSoftScore) cm.getScore());
+        }
+    }
+
+    public IndictmentView(List<Rules> ruleMatched, List<HardSoftScore> score) {
+        this.ruleMatchedList = ruleMatched;
+        this.scoreList = score;
+    }
+
+    public List<Rules> getRuleMatchedList() {
+        return ruleMatchedList;
+    }
+
+    @JsonSerialize(contentUsing = ScoreJacksonJsonSerializer.class)
+    @JsonDeserialize(contentUsing = HardSoftScoreJacksonJsonDeserializer.class)
+    public List<HardSoftScore> getScoreList() {
+        return scoreList;
+    }
+
+    public void setRuleMatchedList(List<Rules> ruleList) {
+        this.ruleMatchedList = ruleList;
+    }
+
+    public void setScoreList(List<HardSoftScore> scoreList) {
+        this.scoreList = scoreList;
+    }
+
+    public static enum Rules {
+        MISSING_REQUIRED_SKILL,
+        EMPLOYEE_ALREADY_HAS_A_SHIFT_ON_DATE,
+        EMPLOYEE_HAS_SHIFTS_LESS_THAN_1O_HOURS_APART,
+        UNDESIRED_TIMESLOT_FOR_EMPLOYEE,
+        DESIRED_TIMESLOT_FOR_EMPLOYEE,
+        UNAVALIABLE_TIMESLOT_FOR_EMPLOYEE,
+        EMPLOYEE_IS_NOT_ROTATION_EMPLOYEE,
+        UNKNOWN;
+
+        public static Rules forConstraintMatch(ConstraintMatch cm) {
+            switch (cm.getConstraintName()) {
+                case "Required skill for a shift":
+                    return MISSING_REQUIRED_SKILL;
+                case "Unavailable time slot for an employee":
+                    return UNAVALIABLE_TIMESLOT_FOR_EMPLOYEE;
+                case "At most one shift assignment per day per employee":
+                    return EMPLOYEE_ALREADY_HAS_A_SHIFT_ON_DATE;
+                case "No 2 shifts within 10 hours from each other":
+                    return EMPLOYEE_HAS_SHIFTS_LESS_THAN_1O_HOURS_APART;
+                case "Undesired time slot for an employee":
+                    return UNDESIRED_TIMESLOT_FOR_EMPLOYEE;
+                case "Desired time slot for an employee":
+                    return DESIRED_TIMESLOT_FOR_EMPLOYEE;
+                case "Employee is not rotation employee":
+                    return EMPLOYEE_IS_NOT_ROTATION_EMPLOYEE;
+                default:
+                    return UNKNOWN;
```

Comment:
Fail fast.

throw IllegalStateException, remove UNKNOWN.
copy the "throw new IllegalStateException(...)" from one of the other switches I already wrote (here or in optaplanner-core), so the exception message is consistent.